bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/config/SnmpChannelConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.snmp.internal.config;
+
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.openhab.binding.snmp.internal.SnmpChannelMode;
+import org.openhab.binding.snmp.internal.SnmpDatatype;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.Variable;
+
+/**
+ * The {@link SnmpChannelConfiguration} class contains fields mapping channel configuration parameters.
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+public class SnmpChannelConfiguration {
+    public String oid;
+    public SnmpChannelMode mode = SnmpChannelMode.READ;
+    public SnmpDatatype datatype;
+
+    public String onvalue;
+    public String offvalue;
+
+    public ChannelUID _channelUID;
+    public OID _oid;
+    public Variable _onValue;
```

Comment:
Please follow naming guidelines, ie do not start with a _ here

Reply:
The reason I chose those names is that it is clear  that these are computed values while the others are real parameters.  Also oid would conflict.  Any sugestion for a not-confusing, namig-guideline compliant naming?

Reply:
I feel this object has two responsibilities, what about converting the framework configuration-object to an object which only contains (immutable) calculated values. In that way it will be better separated and the live-cycle will be much clearer. Do you understand what I mean?

So for example the method `generateChannelConfigs` will not return this object but a much safer immutable object containing only the calculated fields.